File: src/game/tools/rpc/output/wrap/GuildPvPRequestWrap.go
package wraps

import (
	rpc "tipcat.com/common/rpc/simple"
)

type GuildPvPRequestWrap struct {
	
	MemberID int64 // 会员id
	Name string // 名字
	IconID int32 // 头像ID
	Msg string // 留言内容
	rpc.Wrapper
}

func (w *GuildPvPRequestWrap)Decode(pck *rpc.Packet) rpc.Wrapper {
	
	w.MemberID = pck.PopInt64(); 
	w.Name = pck.PopString(); 
	w.IconID = pck.PopInt32(); 
	w.Msg = pck.PopString(); 
	return w
}

func (w *GuildPvPRequestWrap)Encode(pck *rpc.Packet) {
	
	pck.PutInt64(w.MemberID) 
	pck.PutString(w.Name) 
	pck.PutInt32(w.IconID) 
	pck.PutString(w.Msg) 
}

File: src/game/tools/rpc/output/wrap/GuildPvPRequestWrap.cs
using System;
using Common.Net.Simple;

namespace Game 
{
	public class GuildPvPRequestWrap : IWrapper
	{	
		public long MemberID; // 会员id
		public string Name; // 名字
		public int IconID; // 头像ID
		public string Msg; // 留言内容
		
		public void Decode(Packet pck)
		{	
			MemberID = pck.GetLong(); 
			Name = pck.GetString(); 
			IconID = pck.GetInt(); 
			Msg = pck.GetString(); 
		}
		
        public void Encode(Packet pck)
        {	
        	pck.PutLong(MemberID); 
        	pck.PutString(Name); 
        	pck.PutInt(IconID); 
        	pck.PutString(Msg); 
        }
	}
}